### ISSUE TITLE
[FR] Add option to edit default experiment's artifact location

### ISSUE THREAD

**Issue by ampx-mg** [mlflow/mlflow#8700]
### Willingness to contribute

No. I cannot contribute this feature at this time.

### Proposal Summary

Add any possibility whatsoever to edit existing artifact location for the default mlflow experiment with ID 0.

### Motivation

> #### What is the use case for this feature?
When we were first exploring mlflow we used a testing S3 bucket as a artifact location store and then later we switched to mlflow dedicated one. We are now using the dedicated one, but the default experiment still has the original one, which we have meanwhile deleted. Runs in the default experiment now don't have a valid artifact store.

> #### Why is this use case valuable to support for MLflow users in general?
Our users sometimes use the default experiment to just test some stuff or if they forget to set a specific artifact location.

> #### Why is this use case valuable to support for your project(s) or organization?
It is confusing for new users and team members that the default experiment doesn't have an existing artifact store location and they cannot retrieve their artifacts.

> #### Why is it currently difficult to achieve this use case?
I cannot find any way how to achieve this. If possible I would delete the default experiment, but unfortunately that is not possible as mlflow won't allow me to do that.

### Details

We have deployed mlflow using https://artifacthub.io/packages/helm/community-charts/mlflow, we are using mlflow version 2.1.1. If there are any more details needed I can provide them.

Thanks everybody for this amazing tool, we appreciate the hard work!

### What component(s) does this bug affect?

- [X] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by BenWilson2** [comment]
There is an import/export tool that is here: https://github.com/mlflow/mlflow-export-import that is probably what you're looking for. We don't have any plans to enable core functionality within MLflow to migrate runs from one experiment to another. 
Let us know if this works for you :) Thanks for using MLflow!

**Comment by ampx-mg** [comment]
@BenWilson2 Thanks for the reply. I have actually used the import tool before and it is quite handy, but I don't believe it solves the problem.  
I regularly use `gc` from the mlflow cli. If there are runs in the default experiment (with nonexistent S3 bucket as an artifact store) I dont think I can permanently delete them. The command will end with boto3 error message `botocore.errorfactory.NoSuchBucket: An error occurred (NoSuchBucket) when calling the ListObjects operation: The specified bucket does not exist`. I can export all of these runs from default experiment and import them somewhere else, but it seems like an extra step (and with extra tool). I can go back to manually deleting artifacts from S3 and metadata from backend database, which I would rather avoid, but it is my only option at the moment.  
The whole problem could be probably restated even as follows - _delete runs/experiments from mlflow that had their S3 artifact storage deleted_. This is not problem for regular runs/experiments, since I can export them and import them and change the artifact storage, however the default experiment and its runs will just always be there.

**Comment by BenWilson2** [comment]
Yep, acknowledged. We have the default experiment as a 'catch all' safe fallback location so that runs are not 'lost to the ether' or are written to randomly generated experiments if an experiment is not specified. 
We don't have a clear roadmap plan for solving this dilemma, unfortunately. 
As for the gc errors that you're getting, the PR from @PenHsuanWang was merged yesterday and will be part of the next minor release: #8498 so that you won't be suffering through those errors any longer.

**Comment by B-Step62** [comment]
Artifacts for new experiments are fully controllable: pass `artifact_location` to `create_experiment(...)`, or set the server's `--default-artifact-root` (which applies to all newly created experiments). Only the pre-seeded default experiment (id 0) retains its original location; the recommended path is to create a new experiment with the desired location. Closing as not planned, feel free to reopen if you have a use case that a new experiment does not cover.
